autopush/router/apnsrouter.py

```diff
@@ -27,78 +27,129 @@ class APNSRouter(object):
-        """Blocking APNS call to route the notification"""
-        token = router_data["token"]
+        """Blocking APNS call to route the notification
+
+        :param notification: Notification data to send
+        :type notification: dict
+        :param router_data: Pre-initialized data for this connection
+        :type router_data: dict
+
+        """
+        router_token = router_data["token"]
+        self._channel = router_data["channel"]
+        config = self._config[self._channel]
+        apns_client = self.apns[self._channel]
         custom = {
-            "Chid": notification.channel_id,
-            "Ver": notification.version,
+            "chid": notification.channel_id,
+            "ver": notification.version,
         }
         if notification.data:
-            custom["Msg"] = notification.data
-            custom["Con"] = notification.headers["content-encoding"]
-            custom["Enc"] = notification.headers["encryption"]
+            custom["body"] = notification.data
+            custom["con"] = notification.headers["content-encoding"]
+            custom["enc"] = notification.headers["encryption"]
 
             if "crypto-key" in notification.headers:
-                custom["Cryptokey"] = notification.headers["crypto-key"]
+                custom["cryptokey"] = notification.headers["crypto-key"]
             elif "encryption-key" in notification.headers:
-                custom["Enckey"] = notification.headers["encryption-key"]
+                custom["enckey"] = notification.headers["encryption-key"]
 
-        payload = apns.Payload(alert=router_data.get("title",
-                                                     self.default_title),
-                               content_available=1,
-                               custom=custom)
+        payload = apns.Payload(
+            alert=router_data.get("title", config.get('default_title',
+                                                      'Mozilla Push')),
+            content_available=1,
+            custom=custom)
         now = int(time.time())
-        self.messages[now] = {"token": token, "payload": payload}
-        # TODO: Add listener for error handling.
-        self.apns.gateway_server.register_response_listener(self._error)
+        self.messages[now] = {"token": router_token, "payload": payload}
+        apns_client.gateway_server.register_response_listener(self._error)
         self.ap_settings.metrics.increment(
-            "updates.client.bridge.apns.attempted",
+            "updates.client.bridge.apns.%s.attempted" % router_data["channel"],
             self._base_tags)
 
-        self.apns.gateway_server.send_notification(token, payload, now)
+        apns_client.gateway_server.send_notification(router_token, payload,
+                                                     now)
 
         # cleanup sent messages
         if self.messages:
             for time_sent in self.messages.keys():
```

Comment:
It's possible that multiple threads could be iterating on this at once, which means that one of the del calls will throw a KeyError. Would probably be better to toss the message cleanup function back into the event loop so it could clean-up without worry about locking the object.